src/Microsoft.SqlTools.ServiceLayer/EditData/EditTableMetadata.cs

```diff
@@ -55,6 +59,39 @@ public EditTableMetadata()
 
         #endregion
 
+        // Filters out metadata that is not present in the result set, and matches metadata ordering to resultset
+        public static EditColumnMetadata[] FilterColumnMetadata(EditColumnMetadata[] metaColumns, DbColumnWrapper[] resultColumns)
+        {
+            if(metaColumns.Length == 0)
+            {
+                return metaColumns;
+            }
+
+            bool escapeColName = metaColumns[0].EscapedName.Contains("[");
+            Dictionary<string, int> columnNameOrdinalMap = new Dictionary<string, int>(capacity: resultColumns.Length);
+            for (int i = 0; i < resultColumns.Length; i++)
+            {
+                DbColumnWrapper column = resultColumns[i];
+                string columnName =  column.ColumnName;
+                if (escapeColName)
+                {
+                    columnName = columnName.Contains("[") ? columnName : ToSqlScript.FormatIdentifier(columnName);
+                }
+
+                columnNameOrdinalMap.Add(columnName, column.ColumnOrdinal.HasValue ? column.ColumnOrdinal.Value : i);
```

Comment:
As per discussion, let's see what happens if the user uses aliases or has duplicate columns via aliases